src/tasks/BuildBarrel.ts

```diff
@@ -0,0 +1,84 @@
+import { convertPathSeparator } from '../utilities';
+import { QuoteCharacter } from '../options/quoteCharacter';
+import { SemicolonCharacter } from '../options/noSemicolon';
+import { Logger } from '../options/logger';
+import { BaseUrl } from '../options/baseUrl';
+import { loadDirectoryModules } from '../modules';
+import path from 'path';
+import { addHeaderPrefix } from '../builders/header';
+import fs from 'fs';
+import { Directory } from '../interfaces/directory.interface';
+import { buildFileSystemBarrel } from '../builders/fileSystem';
+import { buildFlatBarrel } from '../builders/flat';
+import { StructureOption } from '../options/options';
+
+export const BuildBarrel = ({
```

Comment:
Generally the codebase has `pascalCase` for functions/variables and `CamelCase` for types/interfaces.